odl/discr/discretization.py

```diff
@@ -450,9 +448,7 @@ class DiscretizationVector(RawDiscretizationVector, FnBaseVector):
 
     def __init__(self, space, data):
         """Initialize a new instance."""
-        if not isinstance(space, Discretization):
-            raise TypeError('space {!r} not a `Discretization` '
-                            'instance.'.format(space))
+        assert isinstance(space, Discretization)
```

Comment:
Using asserts in these cases is totally fine IMO, these shouldnt realy be used by users and this makes the code more simple to read.